### ISSUE TITLE
@type annotation doesn't like function object

### ISSUE THREAD

**Issue by NN---** [Microsoft/TypeScript#24280]
The first code sample is not recognized as valid in JS typecheck mode in VSCode.
VS Pro doesn't complain.

Not good:
```js
// @ts-check
/**
 *  @type {{x: function(option: any): void; }}
 */
var y;
```

Good in both:
```js
// @ts-check
/**
 *  @type {{x(option: any): void; }}
 */
var z;
```

**Comment by andy-ms** [comment]
There are two function type syntaxes -- there is the classic JSDoc syntax `function(any): void` and the TypeScript syntax `(option: any) => void`. The first example above is attempting to mix the two syntaxes.

> VS Pro doesn't complain.

That probably indicates that you aren't getting *any* type-checking. (Test with `/** @type {number} */ const x = "";`.) Although I think you should if `// @ts-check` is present.
